pytorch_binding/test/new_benchmark.py

```diff
@@ -0,0 +1,158 @@
+import sys
+import argparse
+import pickle
+
+import torch
+import torch.nn as nn
+from torch.profiler import profile, record_function, ProfilerActivity
+
+T = torch.Tensor
+
+
+class Joint(nn.Module):
+    def __init__(self, hdim: int, V: int) -> None:
+        super().__init__()
+        self.ff = nn.Sequential(
+            nn.Tanh(),
+            nn.Linear(hdim, V)
+        )
+        self.requires_packing = False
+
+    def requires_packing_(self, requires_packing: bool = True):
+        self.requires_packing = requires_packing
+
+    def forward(self, f: T, g: T, lf: T = None, lg: T = None):
+        """
+        f : (N, T, H), output of encoder
+        g : (N, U+1, H), output of predictor
+        lf: (N, ), lengths of seqs in f
+        lg: (N, ), lengths of seqs in g
+        """
+        if self.requires_packing:
+            assert lf is not None
+            assert lg is not None
+
+            H = f.size(-1)
+            # x: (sum(T_i * (U_i+1)), H)
+            x = torch.cat(
+                [
+                    (f[i, :lf[i]].unsqueeze(1) +
+                     g[i, :lg[i]+1].unsqueeze(0)).view(-1, H)
+                    for i in range(f.size(0))
+                ],
+                dim=0
+            )
```

Comment:
Interesting, according to the profiler logs, this is the critical point.

Reply:
move to `benchmark2.py` and add some info.

Reply:
Yes, I guess that's due to there involves lots of slicing operations. To avoid slicing, we could write a custom cuda function something like
```python3
out = custom_packing(f, g, lf, lg)
```
This won't be difficult, but currently, I think it's not in urgent.